src/rpdk/core/fragment/generator.py

```diff
@@ -100,14 +104,12 @@ def __validate_outputs(self, raw_fragments):
         self.__validate_no_exports_present(raw_fragments)
         self.__validate_output_limit(raw_fragments)
 
-    @staticmethod
-    def __get_cfn_lint_matches(raw_fragment):
-        filename = "temporary_fragment.json"
-
-        with open(filename, "w") as outfile:
-            json.dump(raw_fragment, outfile, indent=4, default=str)
-
-        template = cfnlint.decode.cfn_json.load(filename)
+    def __get_cfn_lint_matches(self):
+        filepath = _get_fragment_file(self.fragment_dir)
+        try:
+            template = cfnlint.decode.cfn_json.load(filepath)
+        except json.decoder.JSONDecodeError:
+            template = cfnlint.decode.cfn_yaml.load(filepath)
```

Comment:
Nice!
We could add another `except` for a general exception here. That way if there is any linting failure, we could add a log and display a warning to the user instead of failing the `validate` operation.

Reply:
Addressed this in the latest commit.